### DISCUSSION TITLE
Non-cooperation options

### DISCUSSION THREAD

**Discussion by vteague** [AusOpenTech/ABCData#5] [Ideas]
The scientific literature on data privacy is full of options for deliberate obfuscation in situations like these.
[Ad nauseam](http://ceur-ws.org/Vol-1873/IWPE17_paper_23.pdf) by Howe and Nissenbaum is my favourite example, but there are many.

Like anything else in security and privacy, it depends a lot on your attacker model and security goals.  It's not hard to obstruct a naive data-collection algorithm, but it's hard to defeat a determined attacker without getting your clever plugin blocked.

So here are some imperfect-but-worth-trying examples of non-cooperation that have been suggested to me for dealing with the ABC's data gathering.

1. Share logins, perhaps through a pre-existing service such as [bugmenot](http://bugmenot.com/view/iview.abc.net.au) (no idea why they don't have TLS turned on).

Obviously, this doesn't help against an adversary who looks up the bugmenot page and cancels those accounts.  It's also not clear whether multiple people can use the same account at the same time.

2. Automate the process of generating a new account with each view, perhaps via a browser extension (this was suggested by @eleanor-em). Use the '+' convention so you don't actually need a new email account each time - you just generate a new account with jcitizen+a@gmail.com, then jcitizen+b@gmail.com and so on…

Obviously, this doesn't help against an adversary who ignores the +string suffix when testing whether the email address has been used before. Also, you probably still have to go in to your email and click on the setup link, so it's not fully automated.

These two options could be combined of course: generate a series of +a, +b, ... accounts using a burner email address, then share them on bugmenot.

Any suggestions for improvements, or alternatives that are more robust, are most welcome. Please add them to the discussion here.

**Comment by ghost** [comment]
Email login spoofing is probably not very effective to protect PII as the ABC apk has analytics SDKs and plugins from Facebook, Nielsen, Oztam, Tealium, Snowplow, Youbora and Gigya. These all have their own get device ID / advert ID / user ID routines. ie "Youbora collects data about your viewers as they interact with your videos and more detail about them through your other connected marketing channels."

**Reply by vteague** [reply]
This is a very good point - I'd been thinking about watching from a web browser (which is what I do), but agree that the above ideas do not work for those who use the app, unfortunately.

**Reply by vteague** [reply]
OK, next idea: we could replicate the main idea of AdNauseam, and improve the app with random clicks on lots of different shows (presumably to be run in background when the person wasn't watching).
- upsides: obscures profiling without re-using logins;
- downsides: lots of extra data usage; not sure how technically challenging to make it work; 

It doesn't withhold from the database the things you actually watched, but it adds a bunch of things you didn't watch...

**Reply by ghost** [reply]
The web browser has a gigya server injection on the abc login page, so i assume thats throughout anywhere that a login is required. ie here from login.abc.net.au:
https://cdns.au1.gigya.com/js/gigya.js?apikey=3_IOO-9cJoOAtokw4zcxO55h3m5nHnCiE2pbQLkD3lMZRpQRxfCMErINbcxyKrXcjj

gigya is owned by SAP who has lots of other acquisitions of interest:
https://www.sap.com/investors/en/investment-story/acquisitions.html

**Reply by vteague** [reply]
So what do you think of forking AdNauseam to click on random iView shows in the browser while you're not watching? It's not perfect, but it's the least-bad idea I've thought of so far...

**Reply by vteague** [reply]
Actually I need to think about whether Trackmenot would be better: http://trackmenot.io/
If I understand rightly, the idea is the same, but AdNauseam clicks on ads identified by an ad blocker, while trackmenot issues random search queries to a search engine.  We sort of want an equivalent thing for clicking on random iview shows...

**Reply by ghost** [reply]
I think the main problem is that the ABC uses 3rd party contractors to do many of its functions in its quest to go "digital". So when they engage an "audience survey" company(s), it gets all of ad-tech with it. So for me the problem is how to avoid this extraneous stuff and just listen/watch/read. ie i use AOSP and FDroid only.
As much as I respect Helen Nissenbaum's Contextual Privacy theories I dont think ad nauseam is a solution. There are too many SDKs in the ABC apps that get enough info simply from the moment of install, ie Facebook SDK.

**Reply by vteague** [reply]
Can you explain AOSP and FDroid? (not familiar, sorry.)

**Reply by ghost** [reply]
On the browser you would need to block :
google tag manager gtm.js
gigya.js
https://deliver.oztam.com.au/api/meter

and youbora:
https://a-fds.youborafds01.com/data?outputformat=json&system=abcprod&pluginVersion=6.7.6-adapterless-js&requestNumber=0.4691385882342549&timemark=1621917917122

oh and a part of jwplayer:
jwplayer.js: GET https://ssl.p.jwpcdn.com/player/v/8.10.3/jwpsrv.js net::ERR_BLOCKED_BY_CLIENT

use case: Vivaldi browser, full default privacy settings + plugins uBlock origin, Privacy Badger and NoScript currently blocks all ad-tech listed and Four Corners episode plays.

**Reply by ghost** [reply]
Android Open Source Project = basic, open source android OS ROM, ie LineageOS, no google apps at all.
F-Droid is the open source privacy respecting android app store

**Reply by vteague** [reply]
Any idea why Firefox blocks some but not all versions of gtm.js?
 
![ABCTrackers](https://user-images.githubusercontent.com/11496380/119441537-9ee5ad00-bd69-11eb-8db1-1efa4b8d796f.png)

**Reply by ghost** [reply]
possibly due to firefox allowing what it thinks are necessary scripts for website functions, see: https://blog.mozilla.org/security/2021/03/23/introducing-smartblock/

this is why i use multiple other plugins

**Comment by vteague** [comment]
So on the browser I think we have two distinct issues:
1. generic (third party) trackers
2. the specific js for uploading what you've watched (as per @wabzqem 's json screenshot).

I assume (1) has generic solutions (though probably only somewhat-successful ones). It occurs to me that I am making (2) much harder than it needs to be - you don't have to actually watch the shows (not even via your browser when you're not actually watching), you just have to intercept the json upload and modify it, right? I wonder how hard that is.

**Reply by ghost** [reply]
1. can be blocked by a proper browser with necessary plugins. Behaviour depends on ABC domains when scripts fail to load or timeout.
2. jwplayer requires continuous timestamping as video files are streamed in mp4 segments. It *should* still function without the ad-tech garbage.

**Reply by vteague** [reply]
Hm. So I guess that upload about the things you have actually watched is therefore unavoidable.  Which brings me back to an adnauseam-inspired plugin that pretends to watch other things too.  Not ideal, but at least somewhat obscuring. I wonder if it's possible to do so without actually downloading the videos?

**Comment by ghost** [comment]
Also, google search for iview,then click the first result (Ad) creates this server captured url :
https://iview.abc.net.au/?&WT.srch=1&WT.mc_id=Corp_TV%7cAdWords:%2Babc%20%2Biview_b_g_339709145936_&gclid=EAIaIQobChMIy7X87ZTm8AIVRJVLBR0cfQOkEAAYASAAEgKBRvD_BwE&gclsrc=aw.ds

**Comment by trissylegs** [comment]
I'd ask someone to check on iOS with all the new Privacy features. To see what you can prevent. Of course if have to login they still get a lot.

**Reply by vteague** [reply]
Good idea - at the moment I haven't looked at the app at all, but I too would be very interested to know how it sits with Apple's privacy regime.

**Comment by king-millez** [comment]
[My tweet](https://twitter.com/King_Millez/status/1404654912454029315) 

I don't think "restreaming" is an issue. As long as the playlists are able to be served cross-origin, then you can watch shows on iview outside of iview, tracker and account free.

If playlists return a 403 if a user is not logged in, then an access token must be provided, either in a query string or through a cookie. I can hack together a Node implementation of this today or tomorrow and link it here.

[The app](https://github.com/OpenGovAus/iview-proxy)

**Reply by king-millez** [reply]
Working on this at the moment, even the search bar on iView is riddled with analytics:

```json
{
	"requests": [
		{
			"indexName": "ABC_production_iview_web",
			"params": "highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&getRankingInfo=true&clickAnalytics=true&analytics=true&userToken=anonymous-60451354-4525-401b-9a54-3e18fa480cd5&facetFilters=playable%3A%20-false&query=s&facets=%5B%5D&tagFilters="
		},
		{
			"indexName": "ABC_production_iview_web",
			"params": "highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&getRankingInfo=true&clickAnalytics=true&analytics=true&userToken=anonymous-60451354-4525-401b-9a54-3e18fa480cd5&facetFilters=playable%3A%20-false&query=s&hitsPerPage=6&filters=docType%3A%20category%20OR%20docType%3A%20channel&ruleContexts=iview_categories&facets=%5B%5D&tagFilters="
		},
		{
			"indexName": "ABC_production_iview_web",
			"params": "highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&getRankingInfo=true&clickAnalytics=true&analytics=true&userToken=anonymous-60451354-4525-401b-9a54-3e18fa480cd5&facetFilters=playable%3A%20-false&query=s&hitsPerPage=8&filters=docType%3A%20Program&ruleContexts=iview_programs&page=0&facets=%5B%5D&tagFilters="
		},
		{
			"indexName": "ABC_production_iview_web",
			"params": "highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&getRankingInfo=true&clickAnalytics=true&analytics=true&userToken=anonymous-60451354-4525-401b-9a54-3e18fa480cd5&facetFilters=playable%3A%20-false&query=s&hitsPerPage=8&filters=docType%3A%20VideoEpisode%20AND%20NOT%20subType%3A%20feature%20AND%20NOT%20subType%3A%20livestream&ruleContexts=iview_episodes&page=0&facets=%5B%5D&tagFilters="
		}
	]
}
```

This is just for searching the char "s"

**Reply by king-millez** [reply]
Luckily, you can remove all trackers and it works perfectly fine:

```json
{
        "requests": [
            {
                "indexName": "ABC_production_iview_web",
                "params": "query=doctor who or something"
            }
        ]
    }
```

**Reply by king-millez** [reply]
Interestingly, the search response includes the `m3u8` playlists for shows.

**Reply by king-millez** [reply]
![image](https://user-images.githubusercontent.com/26447912/122058848-677d9400-ce2f-11eb-97fd-5700a2791376.png)
Well it works

**Comment by king-millez** [comment]
SBS on demand content does not require a login to access the content (which is also not locked behind DRM like WideVine):
[https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/sbs.py](https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/sbs.py)

**Reply by king-millez** [reply]
This makes me think that the ABC would rather just put a client side block on content, rather than a server side one. It'd be much easier from their perspective.

**Reply by ghost** [reply]
aight, rolled your node app into an apk and it works. not really sure what to do with it now. will make a release for it at least,  but i doubt its going to end up on fdroid until we see what is happening with the logins. also its currently a real low brow app, no proper ui checks etc. min API 22 ;)
https://github.com/kaputnikGo/AndIViewProxy

**Reply by vteague** [reply]
This looks terrific @king-millez and @kaputnikGo ! 

I think you might have independently rediscovered the same idea as [youtube-dl](https://youtube-dl.org/)
I haven't tried it myself but the idea looks very much the same - thanks to @chrisculnane for pointing it out. It claims to support ABC iview in its [list of supported sites](https://github.com/ytdl-org/youtube-dl/blob/master/docs/supportedsites.md) but of course I'm not sure that will survive past the imposition of compulsory logins.  (But perhaps it won't matter.)

**Reply by king-millez** [reply]
> I think you might have independently rediscovered the same idea as youtube-dl

Sort of, I use youtube-dl regularly. The difference is streaming vs archiving, in my case, this app allows direct searching and streaming of iview content, whereas yt-dl is mainly for downloading and storing content.

**Reply by vteague** [reply]
Huh, terrific, thanks for the clarification.  Both are useful, but I agree streaming is a nice feature for those of us not organised in advance.

Great work on the app @king-millez

**Reply by king-millez** [reply]
Thank you! It looks a bit trash at the moment so give me a day or 2 to review the latest pull request and make styling changes. Then we play the waiting game.

**Comment by ghost** [comment]
On the android iview app there is this package au.net.abc.seesawsdk
which seems to be an ABC in-house analytics SDK for user profile and video tracking separate to the embedded corporate trackers and duplicating the most basic privacy concerns. 

The URL for datasets is https://api.seesaw.abc.net.au/v2/dataset/profile

and the profile dataset contains :
    avatar
    avatarAccessibilityText
    avatarId
    birthYear
    channelFilter
    displayName
    isChild

There are currently 12 trackers in their app, all with access to the same basic info and more, especially device info.
The latest Exodus Privacy report is here:
https://reports.exodus-privacy.eu.org/en/reports/182856/
